server/config/config.go

```diff
@@ -541,6 +593,42 @@ func (cc *RedisClientConfig) String() string {
 	return cc.SimpleTarget
 }
 
+func (c *Configurator) ReconcileFlagsAndConfig() {
+	flagSet := flag.NewFlagSet("", flag.ContinueOnError)
+	defineFlagsForMembers([]string{}, reflect.ValueOf(c.gc).Elem(), flagSet)
+
+	flagSet.VisitAll(func(flg *flag.Flag) {
+		if configSlice, ok := flg.Value.(*flagutil.StringSliceFlag); ok {
+			if flagSlice, ok := flag.Lookup(flg.Name).Value.(*flagutil.StringSliceFlag); ok {
+				originalSlice, ok := originalStringSlices[flg.Name]
+				if !ok {
+					originalStringSlices[flg.Name] = (*flagSlice)[:]
+					originalSlice = originalStringSlices[flg.Name]
+				}
+				// string slices from flags are appended to the values in the config, as
+				// opposed to either overriding the other, so no conflict check is
+				// necessary. Note that Set for StringSliceFlag is performing the
+				// aforementioned append operation, as opposed to setting anything.
+				flg.Value.Set(strings.Join(originalSlice, ","))
+				*flagSlice = *configSlice
+				return
+			}
+			log.Warningf("yaml defines %s as []string, but flags do not.", flg.Name)
+		}
+		_, setInFlags := originalSetFlags[flg.Name]
+		_, presentInYaml := c.presenceMap[flg.Name]
+		// If the flag was set on the command line or there is no value specified in
+		// the config, we use the value defined by the flags. Otherwise (which is to
+		// say, if there was no flag explicitly set on the command line and there is
+		// a value present in the config), use the config value.
```

Comment:
I think it's important to mention that setting the flag here indirectly updates the value within the config struct.